crates/turborepo-lib/src/process/child.rs

```diff
@@ -432,11 +706,65 @@ impl Child {
     }
 }
 
+impl ChildStateManager {
```

Comment:
This struct is exclusively here to reduce the amount of code in `tokio::select` macro as it confuses `rust-analyzer` and even `rustc` making for poor development experience.